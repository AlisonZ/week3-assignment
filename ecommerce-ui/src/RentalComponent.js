import React, {Component} from 'react';
import PropTypes from 'prop-types';

class RentalComponent extends Component {
    static propTypes = {
            title: PropTypes.string,
            houseType: PropTypes.string,
            image: PropTypes.string,
            cost: PropTypes.number,
            stars: PropTypes.number,
            reviews: PropTypes.number,
            isSuperhost: PropTypes.bool,
    }

    render() {
        const houseType = this.props.houseType.toUpperCase();

        return (
            <div>
                <img src={this.props.image} className="rental-image" alt="rental-property" />
                <div className="rental-info">

                    <div className="light-grey-secondary-title">{houseType}</div>
                    <div className="dark-grey-primary-title">{this.props.title}</div>
                    <div className="dark-grey-terciary">${this.props.cost}/night</div>
                    <div> 
                        <span className="dark-grey-four">{this.props.stars} stars </span>
                        <span className="dark-grey-four" >{this.props.reviews}</span>
                        {
                            this.props.isSuperhost ?
                            <span className="dark-grey-four">
                                <span className="bullet-separator">âˆ™</span>
                                <span >Superhost</span>
                            </span > : null
                        } 
                    </div>
                </div>
            </div>
        );
    }
}

export default RentalComponent;