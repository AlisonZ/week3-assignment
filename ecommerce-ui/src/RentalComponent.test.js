import React from 'react';
import {shallow} from 'enzyme';
import RentalComponent from './RentalComponent';
import {MOCK_TITLE, MOCK_HOUSETYPE, MOCK_IMAGE, MOCK_COST, MOCK_STARS, MOCK_REVIEWS, MOCK_IS_SUPERHOST} from './TestMocks';

it('shallow renders Rental Component', () => {
    const wrapper = shallow(
        <RentalComponent
            title={MOCK_TITLE}
            houseType={MOCK_HOUSETYPE}
            image={MOCK_IMAGE}
            cost={MOCK_COST}
            stars={MOCK_STARS}
            reviews={MOCK_REVIEWS}
            isSuperhost={MOCK_IS_SUPERHOST}
         />
    );


    expect(wrapper).toMatchSnapshot();
});