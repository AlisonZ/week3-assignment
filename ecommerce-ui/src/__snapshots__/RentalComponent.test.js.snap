// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders Rental Component 1`] = `
<div>
  <img
    alt="rental-property"
    className="rental-image"
    src="https://via.placeholder.com/350"
  />
  <div
    className="rental-info"
  >
    <div
      className="light-grey-secondary-title"
    >
      APARTMENT
    </div>
    <div
      className="dark-grey-primary-title"
    >
      The Best Place!!
    </div>
    <div
      className="dark-grey-terciary"
    >
      $
      16
      /night
    </div>
    <div>
      <span
        className="dark-grey-four"
      >
        4
         stars 
      </span>
      <span
        className="dark-grey-four"
      >
        122
      </span>
      <span
        className="dark-grey-four"
      >
        <span
          className="bullet-separator"
        >
          âˆ™
        </span>
        <span>
          Superhost
        </span>
      </span>
    </div>
  </div>
</div>
`;
